clap_derive/src/item.rs

```diff
@@ -325,7 +327,10 @@ impl Item {
             if name == "short" || name == "long" {
                 self.is_positional = false;
             }
-            self.methods.push(Method::new(name, quote!(#arg)));
+            match kind {
+                AttrKind::Group => self.group_methods.push(Method::new(name, quote!(#arg))),
```

Comment:
Can you add a test where the group's `id` is overridden?  I think that will break and instead overwrite the command's name.

Similarly, we need a `ui-test` ensuring that `#[group(name = "foo")]` fails as `name` isn't a group attribute

Reply:
The `ui-test` exists but not the overridden `id`.

The problem I'm trying to highlight is that we are only setting `group_methods` in one branch and instead setting things on the `Command` in the other branches.

Maybe we should have a `push_group_method` function and make the switch between them higher up.